Makefile

```diff
@@ -1091,6 +1122,7 @@ endif
 # build .pkg
 .PHONY: pkg
 pkg:
+	$(eval export DEVELOPER_ID_APPLICATION DEVELOPER_ID_INSTALLER TEAMID TSH_BUNDLEID TSH_SKELETON)
```

Comment:
Why are these evals needed? The environment variables are being set above, isn't it enough? We don't use eval anywhere else in our makefiles.

Reply:
They need to be exported. If we unconditionally export them, then the `$(shell security ...)` command will be run every time you invoke `make`. If they go into a `ifeq ($OS),darwin)` block, they will run every time you run `make` on darwin even if you are not making a release or running anything that needs the key IDs. This might be ok, but I was concerned about the side-effects and affecting developers doing normal builds.

Doing it this way only exports the variables if those targets are invoked which means the `security` command is only run then.

BUT, I just remembered there is an unconditional `export` further up in the Makefile - I need to check to see if that exports only the things before that or everything. I didn't notice the command trying to run on Linux, but I will double-check that.

BTW I forgot to mention in the PR description that the individual commits stand on their own and have commit messages describing them. I think I mention some of this in the relevant commit message.

Reply:
I will comment this in the Makefile

Reply:
It looks like this does not work on Linux due to the top-level `export` directive. I will need to add an unexport for those names so it does not unconditionally run the `security find-identity` command to look up the key id. But I will still keep the `$(eval export ...)` as that should only be done for certain recipes.

Reply:
I've fixed that and checked it does not try to run the `security` command on linux or for other mac targets.